### PR TITLE
Added shipment total weight to rate.

### PR DESCRIPTION
**Pull request by rynner**

### CODE REVIEW

File: src/Rate.php [hunk 1]
```diff
@@ -190,6 +190,16 @@ private function createRequest(RateRequest $rateRequest)
         if (isset($deliveryTimeInformation)) {
             $shipmentNode->appendChild($deliveryTimeInformation->toNode($xml));
         }
+          
+        $ShipmentTotalWeight = $shipment->getShipmentTotalWeight();
+        if (isset($ShipmentTotalWeight)) {
+            $shipmentNode->appendChild($ShipmentTotalWeight->toNode($xml));
+        }
+        
+        $InvoiceLineTotal = $shipment->getInvoiceLineTotal();
```
Comment (stefandoorn) [inline src/Rate.php]: This one seems unrelated to `shipmentTotalWeight`. Is this intended?

File: src/Entity/ShipmentTotalWeight.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        $this->UnitOfMeasurement = $var;
+    }
+
+    public function getWeight()
+    {
+        return $this->Weight;
+    }
+
+    public function setWeight($var)
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Could you add phpdoc blocks here?

File: src/Entity/ShipmentTotalWeight.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        return $this->UnitOfMeasurement;
+    }
+
+    public function setUnitOfMeasurement($var)
+    {
+        $this->UnitOfMeasurement = $var;
+    }
+
+    public function getWeight()
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Docblock missing.

File: src/Entity/ShipmentTotalWeight.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        
+        return $node;
+    }
+
+    public function getUnitOfMeasurement()
+    {
+        return $this->UnitOfMeasurement;
+    }
+
+    public function setUnitOfMeasurement($var)
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Should $var be UnitOfMeasurement class? If so, could you add a type hint and also adjust the $var name to something like $unitOfMeasurement?

File: src/Entity/ShipmentTotalWeight.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        if ($this->getUnitOfMeasurement()) {
+            $node->appendChild($this->getUnitOfMeasurement()->toNode($document));
+        }
+
+        $node->appendChild($document->createElement('Weight', $this->getWeight()));
+        
+        return $node;
+    }
+
+    public function getUnitOfMeasurement()
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Docblock missing.

File: src/Entity/Shipment.php [hunk 6]
```diff
@@ -98,6 +98,11 @@ class Shipment
      * @var InvoiceLineTotal
      */
     private $invoiceLineTotal;
+    
+    /**
+     * @var ShipmentTotalWeight
+     */
+    private $ShipmentTotalWeight;
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Could you let this one start lowercase?

File: src/Rate.php (hunk 1 above)
Comment (rynner) [reply src/Rate.php]: Hey sorry. I think the title is a little misleading. I was trying use a Shoptimeintrasnsit request to PR and got error 111546 and then 111549, which lead me to this page:

I had to add both unit of measure and shipment total time to get it to work.

https://developerkitcommunity.ups.com/index.php/Special:AWCforum/st/id449474/

File: src/Entity/Shipment.php [hunk 7]
```diff
@@ -95,9 +95,14 @@ class Shipment
     private $movementReferenceNumber;
 
     /**
-     * @var InvoiceLineTotal
+     * @var invoiceLineTotal
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Class typehint should be exactly the same as class name, probably `InvoiceLineTotal`.

File: src/Entity/Shipment.php [hunk 8]
```diff
@@ -95,9 +95,14 @@ class Shipment
...
      */
     private $invoiceLineTotal;
+    
+    /**
+     * @var shipmentTotalWeight
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Same here, typehint probably wrong name.

File: src/Entity/Shipment.php [hunk 9]
```diff
@@ -544,4 +549,20 @@ public function setDeliveryTimeInformation(DeliveryTimeInformation $deliveryTime
     {
         $this->deliveryTimeInformation = $deliveryTimeInformation;
     }
+    
+    /**
+     * @return shipmentTotalWeight
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Same here, return type class name.

File: src/Entity/Shipment.php [hunk 10]
```diff
@@ -544,4 +549,20 @@ public function setDeliveryTimeInformation(DeliveryTimeInformation $deliveryTime
...
+     */
+    public function getShipmentTotalWeight()
+    {
+        return $this->shipmentTotalWeight;
+    }
+
+    /**
+     * @param shipmentTotalWeight $shipmentTotalWeight
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Wrong class name.

File: src/Entity/Shipment.php [hunk 11]
```diff
@@ -544,4 +549,20 @@ public function setDeliveryTimeInformation(DeliveryTimeInformation $deliveryTime
...
+     */
+    public function setShipmentTotalWeight($shipmentTotalWeight)
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Can you add a type hint here?

```
public function setShipmentTotalWeight(ShipmentTotalWeight $shipmentTotalWeight)
```

File: src/Entity/ShipmentTotalWeight.php [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+namespace Ups\Entity;
+
+use DOMDocument;
+use DOMElement;
+use Ups\NodeInterface;
+use Ups\Entity\UnitOfMeasurement;
+
+class ShipmentTotalWeight implements NodeInterface
+{
+    private $unitOfMeasurement;
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Typehint missing, `UnitOfMeasurement` I think.

File: src/Entity/ShipmentTotalWeight.php [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+    private $Weight;
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Typehint missing, not sure which one it should be.

File: src/Entity/ShipmentTotalWeight.php [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+    public function __construct($response = null)
+    {
+        if (null !== $response) {
+            if (isset($response->unitOfMeasurement)) {
+                $this->setUnitOfMeasurement($response->unitOfMeasurement);
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: This won't work. I think it should be something like:

```
$this->setUnitOfMeasurement(new UnitOfMeasurement($response->unitOfMeasurement));
```

File: src/Entity/ShipmentTotalWeight.php [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+            }
+            if (isset($response->Weight)) {
+                $this->setWeight($response->Weight);
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Not sure this one is correct.

File: src/Entity/ShipmentTotalWeight.php [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+            $node->appendChild($this->getUnitOfMeasurement()->toNode($document));
+        }
+
+        $node->appendChild($document->createElement('Weight', $this->getWeight()));
+        
+        return $node;
+    }
+
+    /**
+     * @return UnitOfMeasurement unitOfMeasurement
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Should be something like:

```
* @return UnitOfMeasurement
```

File: src/Entity/ShipmentTotalWeight.php [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * @param UnitOfMeasurement $unitOfMeasurement
+     */
+    public function setUnitOfMeasurement(UnitOfMeasurement $unitOfMeasurement)
+    {
+        $this->unitOfMeasurement = $unitOfMeasurement;
+    }
+
+    /**
+     * @return weight
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: `string`?

File: src/Entity/Shipment.php (hunk 7 above)
Comment (rynner) [reply src/Entity/Shipment.php]: The class name is capitalized and I have it capitalized in shipment. I'm not sure what your asking for here.

File: src/Entity/Shipment.php (hunk 8 above)
Comment (rynner) [reply src/Entity/Shipment.php]: The class name is capitalized and I have it capitalized in shipment. I'm not sure what your asking for here.

File: src/Entity/ShipmentTotalWeight.php (hunk 15 above)
Comment (rynner) [reply src/Entity/ShipmentTotalWeight.php]: It should be. It's taken from a prior entity (src/shipmentweight)

File: src/Entity/ShipmentTotalWeight.php (hunk 13 above)
Comment (rynner) [reply src/Entity/ShipmentTotalWeight.php]: Not sure what your asking for me here. No other entities has type hinting with the class properties.

**Comment by kevinantonacci** [comment]
Thank you for the great package. Is there anything else holding this back? This issue is preventing the retrieval of international rates.

**Comment by rynner** [comment]
I don't think so. We've been using these changes for a couple months now just fine. Was there changes that were requested that I didn't make?

**Comment by kevinantonacci** [comment]
The changes are working great for me as well. Was just curious when this would be merged into the master branch. Thanks!

**Comment by rynner** [comment]
@kevinantonacci I'm not sure. From what I remember (it's been awhile since I've worked on this) changes were requested, I made them, and it was reviewed again. This went back and forth a few times. Eventually after the last set of changes they never reviewed it or merged it if it was good.

File: src/Entity/Shipment.php [hunk 18]
```diff
@@ -544,4 +549,20 @@ public function setDeliveryTimeInformation(DeliveryTimeInformation $deliveryTime
...
+     */
+    public function getShipmentTotalWeight()
+    {
+        return $this->shipmentTotalWeight;
+    }
+
+    /**
+     * @param ShipmentTotalWeight $shipmentTotalWeight
+     */
+    public function setShipmentTotalWeight($shipmentTotalWeight)
```
Comment (stefandoorn) [inline src/Entity/Shipment.php]: Missing type hint: `public function setShipmentTotalWeight(ShipmentTotalWeight  $shipmentTotalWeight)`

File: src/Entity/ShipmentTotalWeight.php (hunk 12 above)
Comment (stefandoorn) [reply src/Entity/ShipmentTotalWeight.php]: This one needs a docblock, something like:

```
/**
   * @var UnitOfMeasurement
**/
```

File: src/Entity/ShipmentTotalWeight.php [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+use Ups\NodeInterface;
+use Ups\Entity\UnitOfMeasurement;
+
+class ShipmentTotalWeight implements NodeInterface
+{
+    private $unitOfMeasurement;
+
+    private $Weight;
+
+    public function __construct($response = null)
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Missing docblock.

File: src/Entity/ShipmentTotalWeight.php [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+use DOMDocument;
+use DOMElement;
+use Ups\NodeInterface;
+use Ups\Entity\UnitOfMeasurement;
+
+class ShipmentTotalWeight implements NodeInterface
+{
+    private $unitOfMeasurement;
+
+    private $Weight;
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Missing docblock.

File: src/Entity/ShipmentTotalWeight.php [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * @return UnitOfMeasurement
+     */
+    public function getUnitOfMeasurement()
+    {
+        return $this->unitOfMeasurement;
+    }
+
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Blank line too much

File: src/Entity/ShipmentTotalWeight.php [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * @return string weight
+     */
+    public function getWeight()
+    {
+        return $this->Weight;
+    }
+    
+    /**
+     * @param number $weight
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: `number` should be either `int` or `string` and same hint used accordingly in docblock / return hint

File: src/Entity/ShipmentTotalWeight.php (hunk 20 above)
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: Please adjust to `$weight` (lower case starting letter)

File: src/Entity/ShipmentTotalWeight.php [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * @param UnitOfMeasurement $unitOfMeasurement
+     */
+    public function setUnitOfMeasurement(UnitOfMeasurement $unitOfMeasurement)
+    {
+        $this->unitOfMeasurement = $unitOfMeasurement;
+    }
+
+    /**
+     * @return string weight
```
Comment (stefandoorn) [inline src/Entity/ShipmentTotalWeight.php]: `@return string` is enough here

File: src/Rate.php (hunk 1 above)
Comment (stefandoorn) [reply src/Rate.php]: Ok, but can you take this one out and put it in a separate PR? The thing is that the commit should reflect the change and should preferably be a single change instead of combined changes :)

**Comment by rynner** [comment]
Ok I think I made those changes!

File: src/Entity/ShipmentTotalWeight.php [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+    /**
+     * @return int
+     */
+    public function getWeight()
+    {
+        return $this->weight;
+    }
+    
+    /**
+     * @param number $weight
```
[the remark this answers is not in the document]
Comment (rynner) [reply src/Entity/ShipmentTotalWeight.php]: I hope I'm replying to the right comment here. I'm sorry but I can't move this to it's own line or else you won't be able to get rates. If we can guarantee both requests will be merged at the same then I can if not then it will be an issue.

 I can attempt to rename the pull request if that would help.

**Comment by rynner** [comment]
Hey sorry I was going to make these changes today. It was just on the back burner to do when I got a chance. Thanks for merging!
